### ISSUE TITLE
Proposing an Guide for Accessibility in Data Science

### ISSUE THREAD

**Issue by malvikasharan** [alan-turing-institute/the-turing-way#2730]
### Summary

* We currently have some chapters that are buried in different guides, and many aspects of accessibility are currently missing
* In discussion with some organisations, we are proposing placement of a new guide for accessibility

### What needs to be done?

- [ ] Scope out the accessibility guide landing page
- [ ] Brainstorm ideas for different chapters that can be added in their
- [ ] Identify current chapters that need to be moved over

### Who can help?

* @Arielle-Bennett and I have been working with GCHQ to get their draft on accessibility included in The Turing Way
* Anyone from the community who has lived experience and working experience around accessibility in data science

---

### Updates

*Lorem ipsum dolor sit amet, consectetur adipiscing.*

**Comment by malvikasharan** [comment]
Tagging @AndreaSanchezTapia and @LizHareDogs for their help with the landing page for this guide.

Alt text chapter can be one of them.

**Comment by LizHareDogs** [comment]
This looks great and I'm looking forward to the Bookdash and collaborating with everyone.

I'm curious about what you all think about what falls under "accessibility." I've seen it used in two ways in this context. It can refer to the kinds of accessibility practices that we use specifically related to disability. examples are captioning, alt text, andensuring that contrast and size are usable for people with low vision.

There are other things that we do to be inclusive and enhance participation in things, like making sure that folks with older computers and lower bandwidth have access. In some contexts, this falls under "accessibility" and in others  
It falls in the broader category of inclusion, or DEI.

It would be helpful to know which scope we are talking about here as I think about resources to bring to this writing project.  I think it's all important, but I'm not sure which were talking about because I am a newbie to the Turing Way, a USian, and the impact accessibility has on me personally.

**Comment by AndreaSanchezTapia** [comment]
Hi, @LizHareDogs and @aleesteele , I made a push with a markdown file in the accessibility-guide branch: https://github.com/alan-turing-institute/the-turing-way/tree/accessibility-guide - It's failing all the tests but we can check everything tomorrow. 

The landing page would be here and it's very preliminary, but it's based on what we have discussed:
 https://github.com/alan-turing-institute/the-turing-way/blob/accessibility-guide/book/website/accessibility-guide/accessibility-guide.md

**Comment by AndreaSanchezTapia** [comment]
The link for the commit is here: https://github.com/alan-turing-institute/the-turing-way/commit/fc17a315f632a77cb7ede5d4dd73a1b28a8be9ed

**Comment by malvikasharan** [comment]
Tagging Tom @tv074563, Jess @jo77419, and Sam @sspillard.
Here is the HackMD with the draft with content that can be spread out to the landing page that Andrea and Liz are drafting, and one of the first chapters (Overview for instance): https://hackmd.io/@turingway/BJU7IUroj

As a follow-up, we will be planning a 1-day book dash to work on this - planning will begin in February with the onboarding of a new team member in the project.

**Comment by tv074563** [comment]
Thanks for tagging us in, we've been thinking about potential topics and structure. We found it difficult to design the overall structure so have produced a list of topics as a starting point that we think could be overall chapters/sections as follows: 

- Intro
- Aspects of Accessibility
- How Accessibility intersects with Data Science
- Methodologies for Accessible Design
- Visualisation
- Data
- Notebooks
- Web Accessibility/Apps and Dashboards
- Making writing accessible to a non-technical audience
- Accessibility guidance for Talks and Conferences, Training material
- Localisation

I've included a few of the suggestions from our last meeting. It'd be good to decide on a rough outline as that would make it easier to add things in. 

We'd like this to be a practical guide for Data Scientists to easily find relevant guidance for whatever they're aiming to do, so that should probably influence the overall structure. It would be great to talk about this further, maybe at the book dash?

**Comment by trallard** [comment]
Hello all - issue lurker here sharing my 2c.

> I'm curious about what you all think about what falls under "accessibility." I've seen it used in two ways in this context. 

Thanks for raising this question @LizHareDogs. 
My team and I have moved to use the terminology of "access-centred" rather than accessibility as a catch-all, precisely to avoid this context confusion. 

The access-centred term better reflects that the end goal is disability justice and inclusion and implies a radical approach centred around intersectional access. At the same time, accessibility is a practice that falls under that access-centred umbrella and serves as a foundation to build upon.

See Access-centred movement for more details https://accesscenteredmovement.com/what-access-centered-means/

Now onto the topics

> Accessibility for Data Science Products
> Visualisation
> Data
> Notebooks

Some of us (colleagues and I) have been working on these topics. For the last few months, we have been conducting user research/testing with the disabled scientific community. Some of the outputs of this work are authoring guidelines for accessible notebooks. This sounds very much in line with this - so I'd like to explore ways to reuse/collaborate rather than duplicate content (thinking about sustainability and best practices alignment).

**Comment by malvikasharan** [comment]
@trallard Thanks for bringing this up. I had not come across this term before but I think this will actually reduce the confusion and 'specific notion' around what accessibility means. 

I had been largely thinking about accessibility/Access-centred chapters in terms of these three areas:
1. Digital Accessibility in the contexts of:
  *  Design - as listed by you/your team's work and @tv074563 
  * Training and education (relevant contribution from The Carpentries instructor training community)
  * Infrastructure and resources
  * Teams and Collaboration
3. Access based on Geographical contexts
  * Considerations for access from different parts of the world
  * Frugal/low-resource/limited internet technology
  * Considerations for the access in Global South
  * Consideration for different policies that may enhance or limit access 
5. Consideration for Disability and Neurodiversity
  * Some of these are discussed by @LizHareDogs 

These can broadly cover different elements of EDI principles as well as support structures (funding/grants, tools, guidelines, etc.).

**Comment by EstherPlomp** [comment]
Based on a discussion with @LizHareDogs yesterday I went through the book to see where we could link the Accessibility Guide with existing content. My suggestions for the Guide for Reproducible Research are probably more extensive because there is more content and I'm more familiar with it. 

# Guide for Reproducible Research
- Open research should contain a link to the accessibility chapter, probably in the form of open participation since it is relevant to  any of the Open topics. (Open Research should probably still move to the communication guide)
- Version control. Should include some information to make comments more legible.
- Licensing: how to ensure that your licence is machine readable or that you add alt text to license images in your research objects?
- RDM: visualisation and particularly documentation and metadata + sharing and archiving data. Perhaps also mention with FAIR principles (should also add something about CARE..).
- Code Style and Formatting (under Code quality)
- Checklist for code reviewing (Code Reviewing processes)
- Detailed Recommendations for Code Reuse (Reuable Code)
- Research Compendia

# Project Design
- Overview of Project Design
- Personas and Pathways
- File naming convention (capitalise certain things and perhaps other relevant information?)
- Guidelines for Code Styling (Code Styling and Linting)
- Informed consent (Managing Sensitive Data Projects). How to handle informed consent when people can't read/sign a printed form? How to ensure your informed consent form is accessible? 

# Guide for communication 
- Probably a mention everywhere. Maybe not the Binder one?

# Guide for collaboration 
Again, probably everywhere a mention/link. **Exceptions** could be 
- Starting with GitHub
- Research Infrastructure Roles
- Shared Ownership
- Sustainability of Open Source Projects

# Ethical Research 
- Maybe in the Introduction to Research Ethics and Activism for Researchers. The other chapters are perhaps less directly linked. 

# Community Handbook
- Code of Conduct could have a mention
- Style guide can have a mention with further details why certain things are important
- Giving a Turing Way Talk

**Comment by LizHareDogs** [comment]
These are great!

I do think there is room for some comment on accessibility when talking about git and particularly GitHub. GitHub is difficult to navigate for *everybody* whether using a screen reader of not. For people who are used to using the command line, it's a way better way to interact with your repo than navigating the complex screen. 

Accessibility is always enhanced when there is more than one way to do something.

**Comment by andreaczhang** [comment]
There could be a chapter under Communication on **teaching data science/technical skills** to people with no/little IT knowledge, for example medical doctors who want to do their own analysis. At universities there is definitely a high demand of basic data/IT capacity, yet some (or many) universities don't seem to invest enough in training programs in their curriculum. 

The format of The Carpentries is great, and there should be more (only a few sessions per year, long waiting list). I think education and training is very important for reproducible research but unfortunately has been overlooked.

I can help with this chapter (on teaching) if we decide to write one.

**Comment by LizHareDogs** [comment]
Hi and thanks for this terrific idea! I used to work in human psychiatric genetics and I agree!

One point I'd like to make at the intersection with sensory disability is that instruction should be IDE-agnostic because, for example RStudio/Posit is not screen reader accessible.

(* Yes, I know they are advertising that it is "more accessible" now but I've seen no evidence of usability).


Liz Hare, PhD
Dog Genetics LLC
***@***.***
http://www.doggenetics.com <http://www.doggenetics.com/>

**Comment by LizHareDogs** [comment]
Hi all,

I'm looking at the HackMD mentioned above. I'm sorry I'm unable to edit HackMD documents.

The first paragraph, with  its focused on "impairments" and "needs" is not written from a disability rights perspective.

The focus on impairment is related to the deficit model of disability. https://dsq-sds.org/article/view/5236/447

The Deaf community, in particular, rejects being called "hearing impaired."

Framing disabled people as "needy" isn't helpful. The community "needs" to be open to everyone. 
It's not an image that promotes inclusion, collaboration. It encourages a mistaken assumption that disabled people and their needs are "burdens" to everyone else.

**Comment by aleesteele** [comment]
Hi all, just flagging that this draft was reviewed with @LizHareDogs and Laura Ascenzi at Metadocencia. 

We are reviewing the HackMD both in this document: https://hackmd.io/C-LCbjSOQrmEcQl4zlKy-Q?both
as well as this pad: https://pad.sfconservancy.org/p/ttw-guide-accessibility-draft

**Comment by malvikasharan** [comment]
Notes from Tania and Malvika:

- Guide to Access over 'accessibility' will allow broader coverage of concepts and practices
- Content of chapters may include signposting
  - Contextualise by Research lifecycle - at what stage of research cycle are we talking about access 
  - Provide stakeholder type: who is this 'work' serving --> who is engaging (user, contributors, maintainers, end users, citizen etc.)
  - Provide what type of access is this about: Digital, physical, hybrid?
  - Who takes the responsibility for access? - developer, users, leaders, teams (ideally all)
- Turing Way has many accessibility related chapters but they are not findable if people don't specifically look for it
- Many projects exists outside the Turing Way but maybe hard to contextualise for Data Science as people care about specific problem in hand
- The Turing Way handbook and Jupyter (https://github.com/jupyter/accessibility/pull/127 - such as the Authoring guide) has many info that can be generalised and shared for general purpose

**Comment by trallard** [comment]
Hi Liz 

I very much agree - think there was some missing context on that particular point. 

By adding "stage" tags we thought of a way of signaling the "optimal" stage at which a given access tasks/effort should be considered. With the caveat that it can be done or implemented at a later stage but retrofitting or remediating would be like you mentioned more costly or involved. 

For example - adding alt text to your documentation the ideal point to do this would be right when you are writing the content or adding a new image. 
Of course you can do it later (but you might have already negatively impacted some folks' experiences) and will more than likely spend a good amount of time and effort adding or improving the alt text for a bunch of images.


Rather than being used as a way to signal at which point people would need access (because I agree, folks need access at all times).

**Comment by malvikasharan** [comment]
Thanks for adding the context Tania. 
Liz, we totally agree with you. My notes are mostly to structure each chapter as Tania said, stage tag is to ensure that people can browse more easily and get informed about specific practices and consideration based on where they are at in their work.

**Comment by AlexandraAAJ** [comment]
Hello all, 
During the accessibility meeting today we revised these points and we will be working on this in the next couple of months with the aim to present something by the summer. We are organising a monthly session (date to be confirmed and informed in the Slack channel) and coworking spaces to work on this. Here you can find our meeting notes: https://pad.sfconservancy.org/p/ttw-accessibility-wg
